src/ToolsLinux.cpp

```diff
@@ -0,0 +1,44 @@
+#ifdef __unix__
```

Comment:
Not fixed. See my previous comment regarding proper files inclusion in CMAkeLists.txt

Reply:
Yes - you said it wouldn't compile, but it did on my Ubuntu for windows setup.
Is it still an issue if it compiles fine?

Reply:
Because I tried on OS X ;)

Reply:
Actually, I'm getting a compile error since I updated the MoveFile reference.
Fixing now.

Reply:
> There is no need to include this file into a solution if you are on Windows.
I'd agree for the most part, except I feel like on the very odd occasion, I might need to edit the linux or OS X code.

Reply:
Why it is odd? Starcraft can be run under OS X natively and now we have linux version too.
Also I test my bot under OS X only)

Reply:
I think you can manipulate with the solution properties and add linux files to the VS but not still keep it excluded from compilation. I saw such approach several years ago, can't say how it had been done.

Reply:
I've found this, which might work: https://cmake.org/pipermail/cmake/2008-October/024708.html
Also; do you know if the ToolsLinux.cpp code works on OS X as well?
If so, should I renamed it to ToolsUnix.cpp?

Reply:
>Also; do you know if the ToolsLinux.cpp code works on OS X as well?
It works. Yep, you can rename it to Unix if you wish.

Reply:
Done.
I can include a source file for the project but avoid compiling it by using:
```
set_source_files_properties("ToolsUnix.cpp" PROPERTIES HEADER_FILE_ONLY TRUE)
```
But I am hesitant to remove the preprocessor directives however, as VS reacts to it with intelligent syntax highlighting, which I find quite intuitive. I suspect other IDEs would implement similar features as well.
![image](https://user-images.githubusercontent.com/14326310/40717840-00d9f486-644d-11e8-839a-a6b55a91992a.png)

Thus, I suggest either having both (if the cmake stuff helps out in some way) or only use the preprocessor directives.

Reply:
I strongly disagree on this.
1. Some people don't use IDEs.
2. In C++ you should avoid to use the preprocessor as much as possible.
3. There is no real sense in using defines when cmake does all the work needed. This is project-specific thing now.

Reply:
I'm not sure that I agree with your points, but if you feel that strongly about it, I'm fine with going with what you suggested.